src/app/shared/document-hints/document-hints.component.ts

```diff
@@ -94,6 +97,9 @@ export class DocumentHintsComponent implements OnInit, OnChanges {
     if (changes.value) {
       this.filter$.next(this.value);
     }
+    if (changes.collectionId && this.collectionId) {
```

Comment:
Maybe we can test if collectionId changed.

Reply:
`SimpleChanges` object does not contain `collectionId` if the value is not changed. So this condition is `true` only if there is a real change.

Reply:
Good point :)